### PR TITLE
translation works by LiuNing

### PR DESCRIPTION
**Pull request by Ceaser-ln**
1.上传了基础部分搭建神经网络翻译(待修改)
2.构建了文件 basics/06_optimization.md
更新：
basics/06_optimization.md大部分修改完成，我之后会再自行进行细调
更新：
上传了04_build_network.md的大部分翻译
更新：
basics/04_build_network.md基本翻译完成，进行微调中

### CODE REVIEW

File: en/docs/basics/06_optimization.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+So far, we have learned how to use OneFlow to [load data](./03_dataset_dataloader.md),[build models](./04_build_network.md),[calculate gradient of model parameters](./05_autograd.md), and combine them so that we can train models by using backpropagation algorithms.
+
+In [oneflow.optim](https://oneflow.readthedocs.io/en/master/optim.html) ,there are various `optimizer`that simplify the code of back propagation.
+
+This article will first introduce the basic concepts of back propagation and then show you how to use the `oneflow.optim` class.
+
+## numpy 手工实现反向传播
+
+In order to make it easier for readers to understand the relationship between back propagation and automatic derivation, a training process of a simple model implemented only with numpy is provided here:
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:11]:
```suggestion
In order to make it easier for readers to understand the relationship between backpropagation and autograd, a training process of a simple model implemented with numpy is provided here:
```

backpropagation 用作一个词更好？我看 PyTorch Tutorial 是这样做的。用 autograd 更好是因为这是我们系列教程中出现过的内容。这也是为什么说 **先通读系列教程**

only 去掉是因为不必要直译，翻译完成后请主动认真阅读自己的翻译效果，做调整，达到自己认为最佳的效果。

File: en/docs/basics/06_optimization.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+300/500 loss:3.865352482534945e-12
+350/500 loss:3.865352482534945e-12
+400/500 loss:3.865352482534945e-12
+450/500 loss:3.865352482534945e-12
+500/500 loss:3.865352482534945e-12
+w:[[2.000001 ]
+ [2.9999993]]
+```
+
+Note that the loss function expression we selected is  $\sum (y_{p} - y)^2$,so the code for gradient of `loss` to parameter `w` is：
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:70]:
```suggestion
Note that the loss function expression we selected is $\sum (y_{p} - y)^2$, so the code for gradient of `loss` to parameter `w` is：
```

File: en/docs/basics/06_optimization.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+```python
+grad = gradient(x, y, y_pred)
+w -= LR*grad
+```
+
+In summary, a complete iteration in the training includes the following steps:
+
+1. The model calculates the predicted value based on the input and parameters (`y_pred`)
+2. Calculate loss,which is the error between the predicted value and the label
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:87]:
```suggestion
2. Calculate loss, which is the error between the predicted value and the label
```

File: en/docs/basics/06_optimization.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+3. calculate the gradient of loss to parameter
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:88]:
```suggestion
3. Calculate the gradient of loss to parameter(s)
```

File: en/docs/basics/06_optimization.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+4. Update parameter
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:89]:
```suggestion
4. Update parameter(s)
```

File: en/docs/basics/06_optimization.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+1 and 2 are forward propagation process; 3 and 4 are back propagation process.
+
+## Hyperparameters
+
+Hyperparameters are parameters related to model training settings, which can affect the efficiency and results of model training.As in the above code `ITER_COUNT`,`LR` are super parameters.
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:95]:
```suggestion
Hyperparameters are parameters related to model training settings, which can affect the efficiency and results of model training.As in the above code `ITER_COUNT`, `LR` are 【超参】.
```

为什么会把超参翻译诚 super parameters？

File: en/docs/basics/06_optimization.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+## Using the optimizer class in `oneflow.optim`
+
+Using the optimizer class in `oneflow.optim` for back propagation will be more concise. Next, we show how to use it.
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:99]:
```suggestion
Using the optimizer class in `oneflow.optim` for back propagation will be more concise.
```
不一定要直译，比如翻译成英文时，感觉后面的话多余了，就可以去掉。

File: en/docs/basics/06_optimization.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+First, prepare the data and model. The convenience of using Module is that you can place the super parameters in Module for management.
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:101]: 超参

File: en/docs/basics/06_optimization.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+        self.iter_count = iter_count
+
+    def forward(self, x):
+        return flow.matmul(x, self.w)
+
+
+model = MyLrModule(0.01, 500)
+```
+
+### loss function
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:124]: 大小写

File: en/docs/basics/06_optimization.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+
+Then, select the loss function. OneFlow comes with a variety of loss functions. We choose [MSELoss](https://oneflow.readthedocs.io/en/master/nn.html?highlight=mseloss#oneflow.nn.MSELoss) here：
+
+```python
+loss = flow.nn.MSELoss(reduction="sum")
+```
+### Construct optimizer
+The logic of back propagation is encapsulated in optimizer.We choose [SGD](https://oneflow.readthedocs.io/en/master/optim.html?highlight=sgd#oneflow.optim.SGD) here, You can choose other optimization algorithms as needed, such as [Adam](https://oneflow.readthedocs.io/en/master/optim.html?highlight=adam#oneflow.optim.Adam) and[AdamW](https://oneflow.readthedocs.io/en/master/optim.html?highlight=adamw#oneflow.optim.AdamW) .
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:132]: 空格。

The logic of back propagation is encapsulated in optimizer 这句可能不够自然，需要 @wwd190906 @BONNNIE review 时重点确认下是否要改，怎么改

File: en/docs/basics/06_optimization.md [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+
+```python
+optimizer = flow.optim.SGD(model.parameters(), model.lr)
+```
+
+When constructing `optimizer`, pass model parameters and learning rate to `SGD`. Then call `optimizer.step()`, and it automatically completes the gradient of the model parameters and updates the model parameters according to the SGD algorithm.
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:138]: 中文没有主语是自然的。英文大量没有主语会很不自然。

When the `optimizer` is constructed, the model pramters and learning rate is given. Then the `optimizer.step()` is called.

翻译的好坏问题没有特别死板的标准，但是又是客观存在的，需要自己多参考资料，琢磨

File: en/docs/basics/06_optimization.md [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+
+### Train
+
+When the above preparations are completed, you can start training:
```
Comment (doombeaker) [inline en/docs/basics/06_optimization.md:142]: 全文的视角要统一，不要没有特殊理由的情况洗，随意切换 "we" 和 "you"

File: cn/docs/basics/04_build_network.md [hunk 13]
```diff
@@ -132,7 +132,7 @@ print(f"Predicted class: {y_pred}")
 
 `nn.Sequential` 是一种特殊容器，只要是继承自 `nn.Module` 的类都可以放置放置到其中。
 
-它的特殊之处在于：当 Sequential 进行前向传播时，Sequential 会自动地将容器中包含的各层“串联”起来。具体来说，会按照各层加入 Sequential 的顺序，自动地将上一层的输出，作为下一层的输入传递，直到得到整个 Moudle 的最后一层的输出。
+它的特殊之处在于：当 Sequential 进行前向传播时，Sequential 会自动地将容器中包含的各层“串联”起来。具体来说，会按照各层加入 Sequential 的顺序，自动地将上一层的输出，作为下一层的输入传递，直到得到整个 Module 的最后一层的输出。
```
Comment (doombeaker) [inline cn/docs/basics/04_build_network.md:135]: 这个在另外那个PR里已经提交过了。所以不要在这个 PR 中体现。把这个修改恢复吧。

File: en/docs/basics/06_optimization.md [hunk 14]
```diff
@@ -0,0 +1,173 @@
+# Back propagation and optimizer
```
[the remark this answers is not in the document]
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 已修改

File: en/docs/basics/06_optimization.md [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+
+So far, we have learned how to use OneFlow to [load data](./03_dataset_dataloader.md),[build models](./04_build_network.md),[calculate gradient of model parameters](./05_autograd.md), and combine them so that we can train models by using backpropagation algorithms.
```
[the remark this answers is not in the document]
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 均已修改

File: en/docs/basics/06_optimization.md [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+
+In [oneflow.optim](https://oneflow.readthedocs.io/en/master/optim.html) ,there are various `optimizer`that simplify the code of back propagation.
```
[the remark this answers is not in the document]
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 已修改

File: en/docs/basics/06_optimization.md [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+
+This article will first introduce the basic concepts of back propagation and then show you how to use the `oneflow.optim` class.
+
+## numpy 手工实现反向传播
```
[the remark this answers is not in the document]
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 已添加

File: en/docs/basics/06_optimization.md (hunk 1 above)
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 已修改和基本阅读了cn文档

File: en/docs/basics/06_optimization.md (hunk 8 above)
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 已修改

File: en/docs/basics/06_optimization.md (hunk 9 above)
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 已修改

File: en/docs/basics/06_optimization.md (hunk 11 above)
Comment (Ceaser-ln) [reply en/docs/basics/06_optimization.md]: 已修改

File: cn/docs/basics/04_build_network.md (hunk 13 above)
Comment (Ceaser-ln) [reply cn/docs/basics/04_build_network.md]: 已删除
